integration/metamask-init.mjs

```diff
@@ -1,19 +1,50 @@
-import puppeteer from "puppeteer";
-import dappeteer from "dappeteer";
+import dappeteer from "@chainsafe/dappeteer";
+import { addNetwork } from "./addNetwork.mjs";
+import { sleep } from "./utils.mjs";
 
 export const metamaskInit = async () => {
   try {
-    const browser = await dappeteer.launch(puppeteer, {
-      args: ["--no-sandbox"],
-      executablePath: process.env.PUPPETEER_EXEC_PATH,
-      headless: false, // must be false, so can test with metamask extension
-      defaultViewport: null,
-      slowMo: process.argv[2] || 0,
+    const browser = await dappeteer.launch({
+      metaMaskVersion: "v10.31.0",
+      automation: "puppeteer",
+      headless: false,
+      puppeteerOptions: {
+        args: ["--no-sandbox"],
+        executablePath: process.env.PUPPETEER_EXEC_PATH,
+        defaultViewport: null,
+        slowMo: process.argv[2] || 0,
+      },
     });
-    const metamask = await dappeteer.getMetamask(browser, {
+
+    await sleep(2000);
+    const pages = await browser.pages();
+    for (const page of pages) {
+      if (page.url().startsWith("chrome-extension")) {
+        await page.reload();
+      }
+    }
+
+    // Get metamask
+    const metamask = await dappeteer.setupMetaMask(browser, {
       seed: "indicate swing place chair flight used hammer soon photo region volume shuffle",
+      showTestNets: true,
     });
-    await metamask.switchNetwork("localhost");
+
+    // Add network
+    // https://github.com/ChainSafe/dappeteer/blob/b79ab4c74fab87747933d8f428624dcbffc3dd19/test/basic.spec.ts#L117-L119
+    const dappPage = await browser.newPage();
+    await dappPage.goto("https://github.com", { waitUntil: "networkidle" });
+
+    const addNetworkPromise = dappPage.evaluate(addNetwork);
+    try {
+      setTimeout(async () => {
+        await metamask.acceptAddNetwork(true);
+      }, 2000);
+    } catch (e) {
+      // ignore error
+    }
+
+    await addNetworkPromise;
```

Comment:
Not too sure about this section. Should we be using `await sleep(2000)` instead of `setTimeout()`?

Furthermore, the last line can be:

```typescript
await dappPage.evaluate(addNetwork);
```

Reply:
Somehow, if I add the `await`, it won't work... 🤔 

Ill either have to separate the code ([like the Dappeteer testcase](https://github.com/ChainSafe/dappeteer/blob/b79ab4c74fab87747933d8f428624dcbffc3dd19/test/basic.spec.ts#L117-L119)) or just call `dappPage.evaluate(addNetwork)` before `await metamask.acceptAddNetwork(true);`

Anyway I removed the `setTimeout` from the code `await metamask.acceptAddNetwork(true);` as its no longer needed

Reply:
FYI about what the 2 code does, 

`dappPage.evaluate(addNetwork)` opens a page to add the new network as per our configuration

`metamask.acceptAddNetwork(true)` will supposedly click on the button "Approve" to add the new network, and then do a switch of network to the new network ([reference](https://github.com/ChainSafe/dappeteer/blob/fe03f89a5fb494f88cd5778641c8d1d4a831a8f8/src/metamask/addNetwork.ts#L4-L25))